src/server/raft_client.rs

```diff
@@ -201,14 +234,10 @@ impl Buffer for BatchMessageBuffer {
 
     #[inline]
     fn push(&mut self, msg: RaftMessage) {
-        let mut msg_size = msg.start_key.len()
-            + msg.end_key.len()
-            + msg.get_message().context.len()
-            + msg.extra_ctx.len()
-            // index: 3, term: 2, data tag and size: 3, entry tag and size: 3
-            + 11 * msg.get_message().get_entries().len();
-        for entry in msg.get_message().get_entries() {
-            msg_size += entry.data.len();
+        let msg_size = Self::message_size(&msg);
+        // try refresh config before check
+        if let Some(new_cfg) = self.cfg_tracker.any_new() {
```

Comment:
Have you bench the performance impact?

Reply:
In my benchmark, this is no notable impact after this change. In the most common case, this is only one extra atomic load.

Reply:
Actually the atomic load is way heavier than the original that just sums up several numbers. A better place to check for new version is when a message is sent.

Reply:
So do you mean that we move this check into `flush` call as the frequency of flush should be much lower than `push` especially when the lead is heavy.

Reply:
Yes.